enhancements/catalog-data-transfer-reduction.md

```diff
@@ -0,0 +1,450 @@
+1. I am under the impression that a DC is invalid if a bundle in some channel has a replaces property
+that references a bundle without a channel property for that channel,
+since channels are an explicit property of a bundle at the catalog level. Is this true?
+
+## Summary
+
+Disconnected environments suffer from a data transfer bottleneck stemming from in-person ("meatspace")
+image data transfer between connected and disconnected mount points.
+This directly affects operator catalogs, which contain hundreds of packages
+that are updated with new versions constantly; transferring the entire catalog each time
+may require multiple storage media, ex. DVDs, being filled at each transfer event.
+
+The only data that requires transfer is all "new" data, i.e. images published
+since the last transfer. The initial transfer can also be reduced to the packages
+required at cluster creation time, and the latest versions of packages not installed
+for real-time discoverability.
+
+## Motivation
+
+### Goals
+
+#### Reduce the amount of data to download to initially mirror disconnected catalogs
+
+I should be able to prune down a “full” catalog to an input set of packages names,
+package channel names, and package versions (bundles).
+Optionally, dependencies of the remaining catalog items are included.
+
+#### Reduce the amount of data to keep a mirrored catalog updated on regular basis
+
+Given an “old” input set of package names, package channel names, and package versions,
+and an “full” input catalog, I should be able to produce a diff from the point
+of old channel heads in the full catalog. In other words, every reachable bundle
+between the old and new channel heads in the full catalog should be in the diff.
+
+#### Produce a catalog of all channel’s heads from a starting catalog
+
+I should be able to prune down a starting catalog containing one or more
+package channels to only their head bundles. Optionally, dependencies
+of these heads are included. Including heads in catalog output should
+be a mode of other types of pruning/diffing.
+
+#### Provide catalog manipulation functionality as both a command and library
+
+I want to have the option to import functionality described here as a library
+to wrap in some binary, or as a CLI tool that calls the library (`opm`).
+
+#### Define the on disk representation of a differential data set (catalog)
+
+I should be able to configure pruning and diffing catalogs in the same manner,
+and importantly, in a single step, and have an output commonly understood by catalog manipulation tools.
+
+#### Support a simple and controllable approach to updating the disconnected side’s catalog
+
+Once a diff is produced and mirrored, I should be able to add or merge the diff
+with an existing catalog seamlessly. This addition/merge operation should be verified
+to work on the connected side prior to mirroring the diff.
+
+### Non-Goals
+
+1. Perform the mirroring itself.
```

Comment:
I hope this is already covered by the existing `oc adm catalog mirror` or `oc image mirror` tool chain? In the end the user needs to be able to mirror the diff as well.

Reply:
Yes it is, since the diff is a DC. I think @joelanford can speak about that.

Reply:
@joelanford did we already enhance `oc adm catalog mirror` to be able to mirror a DC?

Reply:
Working on it! https://github.com/openshift/oc/pull/868